src/shell/mod.rs

```diff
@@ -1661,7 +1664,10 @@ impl Shell {
             }?;
 
             focus_target = KeyboardFocusTarget::Element(new_target);
-        };
+        } else if let KeyboardFocusTarget::XWaylandGrab(surface) = &focus_target {
+            let new_target = self.element_for_surface(surface)?.clone();
+            focus_target = KeyboardFocusTarget::Element(new_target);
```

Comment:
This doesn't work for override redirect surfaces, no?

(Also won't work with my upcoming fullscreen changes, which just wrap a `CosmicSurface`, but that is unrelated to this PR, more a note to myself, when rebasing.)